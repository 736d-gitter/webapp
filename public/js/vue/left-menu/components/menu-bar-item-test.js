const mount = require('../../__test__/vuex-mount');
const leftMenuConstants = require('../constants');
const { default: MenuBarItem } = require('./menu-bar-item.vue');

describe('menu-bar-item', () => {
  it('all matches snapshot', () => {
    const { wrapper } = mount(MenuBarItem, {
      type: 'all'
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('when all state, all item has highlight matches snapshot', () => {
    const { wrapper } = mount(
      MenuBarItem,
      {
        type: leftMenuConstants.LEFT_MENU_ALL_STATE
      },
      store => {
        store.state.leftMenuState = leftMenuConstants.LEFT_MENU_ALL_STATE;
      }
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it('search matches snapshot', () => {
    const { wrapper } = mount(MenuBarItem, {
      type: leftMenuConstants.LEFT_MENU_SEARCH_STATE
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('people matches snapshot', () => {
    const { wrapper } = mount(MenuBarItem, {
      type: leftMenuConstants.LEFT_MENU_PEOPLE_STATE
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('create matches snapshot', () => {
    const { wrapper } = mount(MenuBarItem, {
      type: leftMenuConstants.LEFT_MENU_CREATE_STATE
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('toggle matches snapshot', () => {
    const { wrapper } = mount(MenuBarItem, {
      type: leftMenuConstants.LEFT_MENU_TOGGLE_STATE
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('calls store action "setLeftMenuState" when item is clicked', () => {
    const { wrapper, stubbedActions } = mount(MenuBarItem, {
      type: leftMenuConstants.LEFT_MENU_ALL_STATE
    });

    wrapper.find({ ref: 'root' }).trigger('click');

    expect(stubbedActions.setLeftMenuState).toHaveBeenCalledWith(
      expect.anything(),
      leftMenuConstants.LEFT_MENU_ALL_STATE,
      undefined
    );
  });

  it('calls store action "toggleLeftMenu" with toggled state when active item is clicked', () => {
    const beforeExpandedState = true;
    const { wrapper, stubbedActions } = mount(
      MenuBarItem,
      {
        type: leftMenuConstants.LEFT_MENU_ALL_STATE
      },
      store => {
        store.state.leftMenuState = leftMenuConstants.LEFT_MENU_ALL_STATE;
        store.state.leftMenuExpandedState = beforeExpandedState;
      }
    );

    wrapper.find({ ref: 'root' }).trigger('click');

    expect(stubbedActions.toggleLeftMenu).toHaveBeenCalledWith(
      expect.anything(),
      !beforeExpandedState,
      undefined
    );
  });

  it('calls store action "toggleLeftMenu" with expanded state when new item is clicked', () => {
    const { wrapper, stubbedActions } = mount(
      MenuBarItem,
      {
        type: leftMenuConstants.LEFT_MENU_ALL_STATE
      },
      store => {
        store.state.leftMenuState = leftMenuConstants.LEFT_MENU_SEARCH_STATE;
        store.state.leftMenuExpandedState = false;
      }
    );

    wrapper.find({ ref: 'root' }).trigger('click');

    expect(stubbedActions.toggleLeftMenu).toHaveBeenCalledWith(expect.anything(), true, undefined);
  });
});
