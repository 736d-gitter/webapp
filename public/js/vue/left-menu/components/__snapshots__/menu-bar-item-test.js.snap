// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu-bar-item all matches snapshot 1`] = `
<button
  aria-label=""
  class="item item-all active"
  type="button"
>
  <div
    class="unread-indicator"
  />
   
  <span
    aria-hidden="true"
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item create matches snapshot 1`] = `
<button
  aria-label=""
  class="item item-create"
  type="button"
>
  <div
    class="unread-indicator"
  />
   
  <span
    aria-hidden="true"
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item people matches snapshot 1`] = `
<button
  aria-label="people"
  class="item item-people"
  type="button"
>
  <div
    class="unread-indicator has-unreads"
  />
   
  <span
    aria-hidden="true"
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item search matches snapshot 1`] = `
<button
  aria-label=""
  class="item item-search"
  type="button"
>
  <div
    class="unread-indicator"
  />
   
  <span
    aria-hidden="true"
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item toggle matches snapshot 1`] = `
<button
  aria-label=""
  class="item item-toggle"
  type="button"
>
  <div
    class="unread-indicator"
  />
   
  <span
    aria-hidden="true"
    class="icon-wrapper"
  />
</button>
`;

exports[`menu-bar-item when all state, all item has highlight matches snapshot 1`] = `
<button
  aria-label="all"
  class="item item-all active"
  type="button"
>
  <div
    class="unread-indicator has-mentions"
  />
   
  <span
    aria-hidden="true"
    class="icon-wrapper"
  />
</button>
`;
